device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1163,7 +1163,15 @@ private void reconnectDeviceSession(TransportException transportException, Strin
         {
             reconnectionAttempts++;
 
-            RetryPolicy retryPolicy = this.getConfig(deviceId).getRetryPolicy();
+            DeviceClientConfig config = this.getConfig(deviceId);
+
+            if (config == null)
+            {
+                log.debug("Reconnection for device {} was abandoned because it was unregistered while reconnecting", deviceId);
```

Comment:
Did you want to add the devicId here in the log of just empty braces?

Reply:
No, this is the correct way for SLF4j log statements to look. For a statement like 

```java
log.debug("device {} has state {} at time {}", deviceId, state, time);
```

it knows to fill in the list of arguments from left to right

```
"device <deviceId> has state <state> at time <time>"
```

Reply:
Nice :)